osdep/timer-win2.c

```diff
@@ -52,7 +81,37 @@ uint64_t mp_raw_time_us(void)
 void mp_raw_time_init(void)
 {
     QueryPerformanceFrequency(&perf_freq);
+
 #if !HAVE_UWP
-    timeBeginPeriod(1); // request 1ms timer resolution
+    // allow (undocumented) control of all the High Res Timers parameters,
+    // for easier experimentation and diagnostic of bug reports.
+    const char *v;
+
+    // 1..1000 ms max timtout for hires (only when MPV_HRT is unset or "auto")
+    if ((v = getenv("MPV_HRT_MAX"))) {
+        int hmax = atoi(v);
+        if (hmax > 0 && hmax <= 1000)
+            hires_max = hmax;
+    }
+
+    // 1..15 ms hires resolution
+    if ((v = getenv("MPV_HRT_RES"))) {
+        int res = atoi(v);
+        if (res >= 1 && res <= 15)
+            hires_res = res;
+    }
+
+    // "always"/"never"/"perwait"  (or "auto" - same as unset)
+    if (!(v = getenv("MPV_HRT")) || !strcmp(v, "auto"))
```

Comment:
Would it make sense for ``MPV_HRT`` and ``MPV_HRT_MAX`` to be combined into a single variable that accepts "auto", "always" and "never" keywords as well as a 1-1000 per-wait value? That would make one less variable to set if the user wants to customize this permanently.

Reply:
> That would make one less variable to set if the user wants to customize this permanently.

I don't think this should be a consideration, and personally I'm not very fond of a variable which could be either a string or a number.

> single variable that accepts "auto", "always" and "never" keywords or a 1-1000 per-wait value

This would not allow to set the resolution in "always" or "auto" modes.

Also, keep in mind that we might want to enhance `MPV_HRT` soon to also support "playback"  (or "perwait+playback" etc).

Reply:
> I don't think this should be a consideration, and personally I'm not very fond of a variable which could be either a string or a number.

Well, mpv is full of such options. It would also make it slightly easier to instruct users to change the value for testing purposes.

> This would not allow to set the resolution in "always" or "auto" modes.

I'm only talking about merging ``MPV_HRT`` and ``MPV_HRT_MAX``. If I understand correctly, that's not the resolution, but the threshold timeout value to use in per-wait mode.

> Also, keep in mind that we might want to enhance MPV_HRT soon to also support "playback" (or "perwait+playback" etc).

This could still be a separate variable (``MPV_HRT_PLAYBACK=yes``.) I think it could also be an mpv option.

Reply:
Come to think of it, I don't mind too much either way. I can see us adding more "conditions" for the hires timer in future that might need an integer argument, so if you think they should stay as separate variables, that's fine by me.

Reply:
> I'm only talking about merging MPV_HRT and MPV_HRT_MAX. If I understand correctly, that's not the resolution, but the threshold timeout value to use in per-wait mode.

You understood correctly, I misread your comment. My bad.

However, you would not be able to set a custom perwait threshold while letting mpv decide between perwait/always depending on the windows version.

> This could still be a separate variable (MPV_HRT_PLAYBACK=yes.) I think it could also be an mpv option.

True for both.

> Come to think of it, I don't mind too much either way.

OK, so mainly due to the reduced functionality (can't describe auto+threshold), I'll stick with the separate values.

If I overlooked something and the functionality is not indeed reduced, then I don't mind much either.

Reply:
> However, you would not be able to set a custom perwait threshold while letting mpv decide between perwait/always depending on the windows version.

I don't really see that as a big use-case, since if someone is setting a custom threshold, they probably want per-wait mode. Maybe if it was an mpv option it would make sense, since someone could be sharing the same config file between Win7 and Win10, but I don't see the same thing happening with env vars. Still, I'm fine either way.

Reply:
>  if someone is setting a custom threshold, they probably want per-wait mode

Agreed, _probably_.

> someone could be sharing the same config file between Win7 and Win10, but I don't see the same thing happening with env vars

Why not? it being env vars rather than option is imposed on the user, if they could do that with at a config file then they would IMO. If I'm sharing a config between my window 7 and 10, including env vars via a batch file wrapper, I wouldn't want to check the windows version myself at the batch file.

> Still, I'm fine either way.

OK, so for the bigger use cases coverage, I'll keep it separate. Thanks.

Reply:
> If I'm sharing a config between my window 7 and 10, including env vars via a batch file wrapper, I wouldn't want to check the windows version myself at the batch file.

Yeah, I guess that's a reasonable case for separate vars.